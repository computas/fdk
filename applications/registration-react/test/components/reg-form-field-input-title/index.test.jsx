import React from 'react';
import { shallow } from 'enzyme';
import InputTitleField from '../../../src/components/reg-form-field-input-title';

let defaultProps;
let wrapper;
let onToggleTitle;
let preventDefault;

beforeEach(() => {
  onToggleTitle = jest.fn();
  preventDefault = jest.fn();
  defaultProps = {
    input: {
      name: 'title.nb',
      value: 'Datakatalog for RAMSUND OG ROGNAN REVISJON'
    },
    meta: {
      active: false,
      asyncValidating: false,
      autofilled: false,
      dirty: false,
      form: 'catalog',
      initial: 'Datakatalog for RAMSUND OG ROGNAN REVISJON',
      invalid: false,
      pristine: true,
      submitting: false,
      submitFailed: false,
      touched: false,
      valid: true,
      visited: false
    },
    label: 'Tittel',
    hideInput: false,
    showLabel: false,
    onToggleTitle
  };
  wrapper = shallow(<InputTitleField {...defaultProps} />);
});

test('should render InputTitleField correctly', () => {
  expect(wrapper).toMatchSnapshot();
});

test('should handle onToggleTitle', () => {
  const mockedEvent = {
    preventDefault
  };
  wrapper.find('button').prop('onClick')(mockedEvent);
  expect(onToggleTitle).toHaveBeenCalled();
});
