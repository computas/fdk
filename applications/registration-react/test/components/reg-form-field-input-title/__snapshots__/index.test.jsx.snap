// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render InputTitleField correctly 1`] = `
<div>
  <div
    className="d-flex align-items-center"
  >
    <button
      className="fdk-edit ml-2 mt-2 nowrap"
      onClick={[Function]}
    >
      <i
        className="fa fa-pencil mr-2"
      />
      Rediger tittel
    </button>
  </div>
</div>
`;
